example/build_cmake_installed_kk_as_language/CMakeLists.txt

```diff
@@ -4,14 +4,19 @@ cmake_minimum_required(VERSION 3.19)
 
 # Projects can safely mix languages - must have C++ support
 # Kokkos flags will only apply to C++ files
-project(Example CXX Fortran CUDA)
+project(Example CXX Fortran)
 
 # Look for an installed Kokkos
 find_package(Kokkos COMPONENTS separable_compilation)
-set_source_files_properties(cmake_example.cpp PROPERTIES LANGUAGE CUDA)
+# Enable languages used by Kokkos
+kokkos_enable_language()
+
+# Mark source files to be compiled in the language that is enabled in Kokkos (if cmake supports that language)
+kokkos_set_source_language(cmake_example.cpp)
 add_executable(example cmake_example.cpp bar.cpp foo.f)
 
-# This is the only thing required to set up compiler/linker flags
+# Set target to use architecture and std that is used in Kokkos
+kokkos_set_target_properties(example)
 target_link_libraries(example Kokkos::kokkos)
```

Comment:
Why at this point not do the target_link_libraries as well inside the kokkos_set_target_properties ...

Reply:
I have no hard feelings about this, I just thought it is more expressive (aka: do as you normally do for libraries, just use our functions to add the backup-agnostic language markups)